### ISSUE TITLE
TiCDC there are metrics for tables that have already been dropped.

### ISSUE THREAD

**Issue by hanbinbin731** [pingcap/tiflow#11073]
### What did you do?

```sql
create table test (a int primary key);
drop table test;
```
```sh
wget http://192.168.1.13:8300/metrics
vi metrics
```

### What did you expect to see?

Metrics for tables that have already been dropped should not exist.

### What did you see instead?

```
ticdc_kvclient_channel_size{changefeed="test-task",namespace="default",table="80",type="err"} 0
ticdc_kvclient_region_worker_busy_ratio{changefeed="test-task",namespace="default",store="192.168.1.11:20160",table="80",type="event-handler"} 0
```

### Versions of the cluster

```
Release Version: v6.5.8
Git Commit Hash: e5999e3c21bee8748126951a869522e182d207b6
Git Branch: heads/refs/tags/v6.5.8
UTC Build Time: 2024-01-25 10:03:56
Go Version: go version go1.19.13 linux/amd64
Failpoint Build: false
```

**Comment by asddongmen** [comment]
Only affect version < v7.5.0.

**Comment by asddongmen** [comment]
fixed by #11184
